vertical-pod-autoscaler/pkg/recommender-external/logic/recommender.go

```diff
@@ -0,0 +1,62 @@
+package logic
+
+import (
+	"k8s.io/klog/v2"
+
+	"k8s.io/autoscaler/vertical-pod-autoscaler/pkg/recommender-external/model"
+	other_logic "k8s.io/autoscaler/vertical-pod-autoscaler/pkg/recommender/logic"
+	other_model "k8s.io/autoscaler/vertical-pod-autoscaler/pkg/recommender/model"
+)
+
+// PodResourceRecommender computes resource recommendation for a Vpa object.
+type PodResourceRecommender interface {
+	// TODO: If we had a way to get the vpaID from AggregateContainerState we could implement the upstream interface instead.
+	GetRecommendedPodResources(containerNameToAggregateStateMap other_model.ContainerNameToAggregateStateMap, containerNameToRecommendedResources model.ContainerNameToRecommendation) other_logic.RecommendedPodResources
+}
+
+type podResourceRecommender struct {
+}
+
+func NewPodResourceRecommender() PodResourceRecommender {
+	return &podResourceRecommender{}
+}
```

Comment:
Compose the `model.ContainerNameToRecommendation` and pass it via the constructor.

Call the constructor each time your need the PodResourceRecommender in the Recommender, it is a stateless object, and building on the fly cost nothing (no cache, no big allocation, it is only carrying logic)

Reply:
done